### PR TITLE
Cleanup and synchronize options in manpages

### LINKED ISSUE #1772: Check manpages changes
It can be useful to add a warning in jenkins if the options of a program in `src/` are changed but the PR does not contain `share/man/` changes.

Reminder for @privat

### PR DESCRIPTION
**Pull request by privat**
A tedious work to detect and fix undocumented options in the manpages of tools.
This also synchronizse the description of `--help` and the text in manpages (when there is a diversion, the best one if often selected).

Cherry on the cake: a new script `check_manpage` can automatically check that new options of tools are documented in the man. Close #1772

### CODE REVIEW

File: misc/jenkins/check_manpages.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	fi
+
+	$bin --help | grep '^  ' | sed 's/    */;/;s/  //' > check_manpages-option_list.out
+	while IFS=';' read opt cmt; do
+		# Generate first lines for the man page from --help
+		printf "%s\n" "$opt" | sed 's/, /`, `/g;s/^/### `/;s/$/`/' > check_manpages-from_help.out
+		echo "$cmt." >> check_manpages-from_help.out
+
+		# Generate grep motif to search in the existing manpage
+		printf "%s\n" "$opt" | sed 's/, /`\\|`/g;s/^/###.*\\(`/;s/$/`\\)/' > check_manpages-grep_motif.out
```
Comment (R4PaSs) [inline misc/jenkins/check_manpages.sh]: :%s/motif/pattern

**Comment by R4PaSs** [comment]
:shipit:

**Comment by privat** [comment]
The script passed: http://gresil.org/jenkins/job/CI_github/3680/testReport/cmd/check/manpages/

**Comment by privat** [comment]
ping @Morriar as he is the author of the issue

**Comment by Morriar** [comment]
This is very cool!

+1

**Pull request merged**